internal/pkg/agent/transpiler/ast.go

```diff
@@ -982,6 +1064,11 @@ func attachProcessors(node Node, processors Processors) Node {
 
 // Lookup accept an AST and a selector and return the matching Node at that position.
 func Lookup(a *AST, selector Selector) (Node, bool) {
+	// Be defensive and ensure that the ast is usable.
+	if a == nil || a.root == nil {
+		return nil, false
+	}
```

Comment:
We should probably write some tests for these kinds of cases. Not necessarily in this PR, but I recall recently adding similar conditions.

Reply:
I can add a test.

Reply:
Test added.